libraries/standard/mqtt/src/mqtt.c

```diff
@@ -99,6 +101,427 @@ static int32_t sendPacket( MQTTContext_t * pContext,
+    {
+        IotLogErrorWithArgs( "Dumped packet. Dumped bytes=%d",
+                             totalBytesReceived );
+        /* Packet dumped, so no data is available. */
+        status = MQTTNoDataAvailable;
+    }
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Receive a packet from the transport interface.
+ *
+ * @param[in] pContext MQTT Connection context.
+ * @param[in] incomingPacket packet struct with remaining length.
+ *
+ * @return MQTTSuccess or MQTTRecvFailed.
+ */
+static MQTTStatus_t receivePacket( MQTTContext_t * const pContext,
+                                   MQTTPacketInfo_t incomingPacket )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    int32_t bytesReceived = 0;
+    int32_t totalBytesReceived = 0;
+    size_t bytesToReceive = 0;
+
+    assert( pContext != NULL );
+
+    if( incomingPacket.remainingLength > pContext->networkBuffer.size )
+    {
+        bytesToReceive = pContext->networkBuffer.size;
+        IotLogErrorWithArgs( "Incoming packet length %u exceeds network buffer size %u."
+                             "Incoming packet will be dumped",
+                             incomingPacket.remainingLength
+                             pContext->networkBuffer );
+    }
+    else
+    {
+        bytesToReceive = incomingPacket.remainingLength;
+    }
+
+    bytesReceived = pContext->transportInterface.recv( pContext->transportInterface.networkContext,
+                                                       pContext->networkBuffer.pBuffer,
+                                                       bytesToReceive );
+    totalBytesReceived = bytesReceived;
+
+    /* Partial receive. */
+    if( bytesReceived < ( int32_t ) bytesToReceive )
+    {
+        IotLogWarnWithArgs( "Packet partially received: Bytes received=%d,"
+                            "Bytes expected=%u",
+                            bytesReceived,
+                            bytesToReceive );
+        /* Handle partial receive. */
+        bytesReceived = pContext->transportInterface.recv( pContext->transportInterface.networkContext,
+                                                           ( pContext->networkBuffer.pBuffer + totalBytesReceived ),
+                                                           ( bytesToReceive - totalBytesReceived ) );
+        totalBytesReceived += bytesReceived;
+        if( totalBytesReceived != ( int32_t ) bytesToReceive )
+        {
+            IotLogErrorWithArgs( "Handling partial receive failed. Bytes "
+                                 "received=%d, Bytes expected=%u",
+                                 totalBytesReceived,
+                                 bytesToReceive );
+            status = MQTTRecvFailed;
+        }
+    }
+    /* Too many bytes received. */
+    else if( bytesReceived > ( int32_t ) bytesToReceive )
+    {
+        IotLogErrorWithArgs( "Too many bytes received. Bytes received=%d, ",
```

Comment:
Log message missing the "Failure-Outcome" section. Suggest using: `Failure in receving MQTT packet: Received more too many bytes from transport recv(): BytesReceived=%d, BytesExpected=%u)`

Reply:
Branch removed as it will never execute